### PR TITLE
Enable auto-publishing stable releases to choco 

### PR DESCRIPTION
**Pull request by marcparadise**
## Description

This adds Chocolatey to the third-party-packages
buildkite pipeline, so that every stable release will
get published to Chocolatey.

### CODE REVIEW

File: .expeditor/publish_to_chocolatey.ps1 [hunk 1]
```diff
@@ -0,0 +1,46 @@
+$ErrorActionPreference="stop"
+
+Write-Host "--- Fetching latest release data from omnitruck API"
+# TODO WARNING DOOM: make sure  v=latest after testing complete
```
Comment (tyler-ball) [inline .expeditor/publish_to_chocolatey.ps1:4]: The strictest of `TODO`s

File: .expeditor/publish_to_chocolatey.ps1 [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+Write-Host "--- Publishing package"
+
+$valid_build_creator="Chef Expeditor"
+$pack_cmd = "choco pack $tempDir/chef-workstation.nuspec --version " + $releaseRecord.version"
+$publish_cmd = "choco push chef-workstation.($releaseRecord.version).nupkg --timeout 600"
+
+try {
+    if($env:BUILDKITE_BUILD_CREATOR -eq $valid_build_creator) {
+        Invoke-Expression $pack_cmd
+        if ($LASTEXITCODE -ne 0) { throw "unable to choco pack" }
```
Comment (tyler-ball) [inline .expeditor/publish_to_chocolatey.ps1:36]: Can you update this to also print out the stdout/stderr from the choco command so we know what went wrong? Or is that already echoed by default?

File: components/packaging/chocolatey/chef-workstation.nuspec [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    <title>Chef Workstation</title>
+    <tags>chefdk chef-dk chef workstation chef-client client development chef-workstation</tags>
+    <summary>Chef Workstation gives you everything you need to get started with Chef. Ad-hoc remote execution, scans and configuration tasks, cookbook creation tools, as well as robust dependency and testing software all in one easy-to-install package.</summary>
+    <description>Chef Workstation gives you everything you need to get started with Chef. Ad-hoc remote execution, scans and configuration tasks, cookbook creation tools, as well as robust dependency and testing software all in one easy-to-install package.</description>
+    <authors>Chef Software, Inc.</authors>
+    <owners>Chef Software, Inc.</owners>
+    <copyright>2018-2020 Chef Software, Inc.</copyright>
+    <requireLicenseAcceptance>false</requireLicenseAcceptance>
+    <bugTrackerUrl>https://github.com/chef/chef-workstation/issues</bugTrackerUrl>
+    <docsUrl>https://www.chef.sh/docs/chef-workstation/getting-started/</docsUrl>
```
Comment (tyler-ball) [inline components/packaging/chocolatey/chef-workstation.nuspec:14]:
```suggestion
    <docsUrl>https://docs.chef.io/workstation/</docsUrl>
```

File: .expeditor/publish_to_chocolatey.ps1 (hunk 2 above)
Comment (marcparadise) [reply .expeditor/publish_to_chocolatey.ps1:36]: The output is included in the logs by default.

**Comment by marcparadise** [comment]
Code review updates are completed.  @tyler-ball I agree that it's something to track; there's also a lag-time for package approval, so we won't be able to verify immediately/programmatically that it's ready to go. 

I have published successfully versions 0.16.33 and 0.15.18, and both are pending review.   I'll keep checking on them, but I'm not expecting problems. 

The only remaining thing to verify requires merging the PR - making sure that the pipeline triggers correctly once the next stable release is published.

**Comment by marcparadise** [comment]
capture_082

**Comment by marcparadise** [comment]
capture_084

**Pull request merged**